File: 프로그래머스/lv1/42748. K번째수/K번째수.java
import java.util.ArrayList;
import java.util.Arrays;

class Solution {
    public ArrayList<Integer> solution(int[] array, int[][] commands) {

        ArrayList<Integer> answer = new ArrayList<>();
       
        for (int[] command : commands) {
            int i = command[0];
            int j = command[1];
            int k = command[2];

            int[] newList = Arrays.copyOfRange(array,i-1,j);
            
            Arrays.sort(newList);
            
            answer.add(newList[k-1]);

        }

        
        return answer;
    }
}

File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

int cal(int to , int from,int at,vector<int> arr) {
    vector<int> sortArr;
    for (;to<=from;to++) {
        sortArr.push_back(arr[to]);
    }
    sort(sortArr.begin(),sortArr.end());

    return sortArr[at];
}
vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    for (vector<int> command : commands) {
        int to = command[0]-1;
        int from = command[1]-1;
        int at = command[2]-1;
        answer.push_back(cal(to,from,at,array));
    }

    return answer;
}